pkg/curl/schemes.go

```diff
@@ -347,7 +346,13 @@ func (s *SchemeWithRetries) Fetch(ctx context.Context, u *url.URL) (io.ReaderAt,
 		if s.DoRetry != nil && !s.DoRetry(u, err) {
```

Comment:
I'm always a little concerned when I see these nil comparison, makes me feel like the design is a bit off, but that's not your code ...

Reply:
A type like Optional[T] would like make it more obvious, but I think the != nil here is used to make this DoRetry an optional value. When nil we retry until success.